kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/config/tls/NettyTrustProvider.java

```diff
@@ -36,6 +38,7 @@ public SslContextBuilder apply(SslContextBuilder builder) {
             public SslContextBuilder visit(TrustStore trustStore) {
                 try {
                     enableHostnameVerification();
+                    enableClientAuthIfRequired(trustStore);
```

Comment:
```suggestion
                    enableClientAuth(trustStore);
```

`IfRequired` is just noise in my book, I'd argue thats up to the implementation of `enableClientAuth` to decide when or when not to enable it. IIRC its the same pattern as `enableHostnameVerification`